qos/evm/request_validator.go

```diff
@@ -0,0 +1,170 @@
+package evm
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	"github.com/buildwithgrove/path/gateway"
+	qosobservations "github.com/buildwithgrove/path/observation/qos"
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+)
+
+// maximum length of the error message stored in request validation failure observations and logs.
+const maxErrMessageLen = 1000
+
+// requestValidator handles EVM request validation, generating appropriate error contexts
+// when validation fails or request contexts when validation succeeds.
+type requestValidator struct {
+	logger        polylog.Logger
+	chainID       string
```

Comment:
How do you decide when we use serviceID vs chainID?

I can argue that since `evm` is definitely a chain, it makes sense. This ties into my point above so think about it w/ appropriate TODOs.